### PR TITLE
Update filter boxes lifecycle

### PR DESCRIPTION
**Pull request by phoebemirman**
We want the filter boxes to show new data and update and re-sort/filter as data gets changed. But we only want to make these updates with certain pieces of data.

Tables only update sorting and filtering under certain circumstances:

1. Positions table should re-sort when the markets change reporting state, most recently traded changes, current value or total returns get updated, or new markets get added/removed to/from this table.

2. Markets are updated when new markets are added/removed, reporting states change, or most recently traded changes. 

3. Filled/Open orders update when the number of items in their lists change, so new markets get filled/open orders in them, or new filled/open orders are made.

Test cases: 
- Open orders
    - When you cancel an order, should show pending
    - Order gets removed if you cancel it from another page
    - When you create an open order on an already there market open order should show
    - Creating open order on a market not there should show up and be sorted in the list

- Positions
   - When a new market is added list should update with that market
   - When current value or total returns changes, list should update and reorder
   - When recently traded updates list should update
   - A market changes reporting state

closes https://github.com/augurproject/augur/issues/1142
closes https://github.com/augurproject/augur/issues/1162
closes https://github.com/augurproject/augur/issues/1185
closes https://github.com/AugurProject/augur/issues/1171

### CODE REVIEW

File: src/modules/orders/selectors/filled-orders.js [hunk 1]
```diff
@@ -24,15 +24,14 @@ function findOrders(filledOrders, accountId, outcomesData, marketsData) {
       let typeOp = type;
 
       const outcomeName =
-        outcomesData[marketId][outcome].name ||
-        outcomesData[marketId][outcome].name;
+        outcomesData[outcome].name || outcomesData[outcome].name;
```
Comment (bthaile) [inline src/modules/orders/selectors/filled-orders.js]: Not sure what this is? same property || same property, if I'm reading this correctly.

File: src/modules/orders/selectors/filled-orders.js [hunk 2]
```diff
@@ -24,15 +24,14 @@ function findOrders(filledOrders, accountId, outcomesData, marketsData) {
...
 
       if (accountId === creator && !foundOrder) {
         typeOp = type === BUY ? SELL : BUY; // marketTradingHistory is from filler perspective
       }
 
       const timestampFormatted = convertUnixToFormattedDate(timestamp);
-      const marketDescription = marketsData[marketId].description;
+      const marketDescription = marketsData.description;
```
Comment (bthaile) [inline src/modules/orders/selectors/filled-orders.js]: doesn't order already have marketDescription and outcomeName?

File: src/modules/portfolio/containers/filled-orders.js [hunk 3]
```diff
@@ -35,21 +35,34 @@ const mapStateToProps = state => {
...
+      outcomesData[item.id],
+      marketsData[item.id]
     )
   }));
 
   /* eslint-disable */
   let allFilledOrders = [];
   marketIds.map(marketId => {
-    const formattedFilledOrders = selectFilledOrders(marketTradingHistory[marketId], account, outcomesData, marketsData);
+    const formattedFilledOrders = selectFilledOrders(marketTradingHistory[marketId], account, outcomesData[marketId], marketsData[marketId]);
```
Comment (bthaile) [inline src/modules/portfolio/containers/filled-orders.js]: if we have marketDescription and outcomeName then would outcomesData and marketsData not be needed?

File: src/modules/portfolio/containers/my-markets.js [hunk 4]
```diff
@@ -12,7 +13,26 @@ import { createTabsInfo } from "modules/portfolio/helpers/create-tabs-info";
 import { createMarketsStateObject } from "modules/portfolio/helpers/create-markets-state-object";
 
 const mapStateToProps = state => {
-  const markets = createMarketsStateObject(selectAuthorOwnedMarkets(state));
+  const createdMarkets = selectAuthorOwnedMarkets(state);
+
+  const marketsPick = createdMarkets.map((
+    market
+  ) =>
+    pick(market, [
```
Comment (bthaile) [inline src/modules/portfolio/containers/my-markets.js]: We'll come back later and refactor lodash.

File: src/modules/orders/selectors/filled-orders.js [hunk 5]
```diff
@@ -24,15 +24,14 @@ function findOrders(filledOrders, accountId, outcomesData, marketsData) {
...
 
       if (foundOrder) {
         foundOrder.trades.push({
```
Comment (bthaile) [inline src/modules/orders/selectors/filled-orders.js]: outcomeName below, what if scalar market? that should be scalarDenomination

**Comment by bthaile** [comment]
Also, in `findOrders` method the order object is building built up, could you save container code by putting outcomeName and marketDescription on the order?

File: src/modules/orders/selectors/filled-orders.js (hunk 1 above)
Comment (phoebemirman) [reply src/modules/orders/selectors/filled-orders.js]: that is left over from when scalar names were done in there. will remove

**Comment by AugurIntegration** [comment]
This pull request **fixes 1 alert** when merging 10e5271386de4513bdc8ad3c068f3d17ea4ffbcd into 0ad4e43cf02e302b0e99f73291cb6b2fa45986f3 - [view on LGTM.com](https://lgtm.com/projects/g/AugurProject/augur-ui/rev/pr-b1dea2a9635d0ceb0cbbe387a8cc4af768d945c0)

**fixed alerts:**

* 1 for Identical operands

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by AugurIntegration** [comment]
This pull request **fixes 1 alert** when merging 6a37554ef582c224e06a7ad26f48027d0b6cb567 into 9ecf600f1efadbc66183051cf3b70bdb9d2e1f6a - [view on LGTM.com](https://lgtm.com/projects/g/AugurProject/augur-ui/rev/pr-8111b452fb504725a8284b2a956f6ed1f5246cdc)

**fixed alerts:**

* 1 for Identical operands

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by AugurIntegration** [comment]
This pull request **fixes 1 alert** when merging cb9b453031f68974654a0de24b50962a4e0a6eaf into 9ecf600f1efadbc66183051cf3b70bdb9d2e1f6a - [view on LGTM.com](https://lgtm.com/projects/g/AugurProject/augur-ui/rev/pr-76fe6170c7a9b838dbad2900a3c6d2230142ffd2)

**fixed alerts:**

* 1 for Identical operands

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**